src/generic/Matrix.jl

```diff
@@ -22,14 +22,21 @@ Return the parent object of the given matrix.
 parent(a::Mat{T}, cached::Bool = true) where T <: NCRingElement =
     MatSpace{T}(a.base_ring, size(a.entries)..., cached)
 
-dense_matrix_type(::Type{T}) where T <: NCRingElement = MatSpaceElem{T}
-
 @doc Markdown.doc"""
-    dense_matrix_type(R::Ring)
-    
-Return the type of matrices over the given ring.
+    dense_matrix_type(::Type{T}) where T<:NCRingElement
+    dense_matrix_type(::T) where T<:NCRingElement
+    dense_matrix_type(::Type{S}) where S<:NCRing
+    dense_matrix_type(::S) where S<:NCRing
+
+Return the type of matrices with coefficients of type `T` respectively
+`elem_type(S)`.
```

Comment:
Hmm, change requests to myself:
- the docstring should make clear that the method taking the *ring element type* is the central one, the others dispatch to it (that's important for implementors)
- `docs/src/matrix.md` should be updated, too
- also `docs/src/matrix_interface.md`
- and I guess while at it, `docs/src/poly_interface.md`'s discussion of `dense_poly_type` should be updated...
- ... and `mpoly_type` be added ...
- ... and also the `*poly_ring_type` methods?

Reply:
I am not merging, because of these comments here.

Reply:
I've updated the `dense_matrix_type` docs, and will open an issue for the remainging bullet points.

Reply:
I assume this means it is good to go?

Reply:
Yeah, I meant to merge it yesterday but apparently forgot